pkg/vm/memory/memory_value.go

```diff
@@ -93,6 +93,12 @@ func (address *MemoryAddress) Sub(lhs *MemoryAddress, rhs any) (*MemoryAddress,
 	}
 }
 
+func (address *MemoryAddress) Relocate(segmentsOffset []int) *f.Element {
+	felt := f.Element{}
+	felt.SetUint64(uint64(segmentsOffset[address.SegmentIndex]) + address.Offset)
```

Comment:
Should we check for the overflow here?

Reply:
For an overflow to occur here the VM should have run with much more than 2**64 bytes of memory (an exabyte of data 😅) I think we need not to worry about this right now

Reply:
Ok, then I don't see the point of having the felt as a returned value. ```SetUint64``` is not a free operation it involves converting into the Montgomery form under the hood which involves quite a few instructions. Can we get away with not calling it here and returning just ```uint64```?

Reply:
If we return uint64, at some point we would have to create a `*f.Element` in order to store it in the final relocated memory. Of course we can create a more efficient memory relocation, but I don't see it without sacrificing a lot of current impl simplicity. Let's have an mvp first ready for benchmarking. That will help us better to understand what is lagging behind, maybe this is not among major issues.

Reply:
Fair enough!